File: app/src/main/java/com/veganbeauty/app/features/profile/AccountProfileAddressFragment.java
package com.veganbeauty.app.features.profile;

import android.content.Context;
import android.graphics.Color;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;

import com.google.android.material.chip.Chip;
import com.google.android.material.chip.ChipGroup;
import com.veganbeauty.app.R;
import com.veganbeauty.app.core.base.RootieFragment;
import com.veganbeauty.app.databinding.AccountProfileAddressBinding;
import com.veganbeauty.app.features.home.BottomNavHelper;
import com.veganbeauty.app.features.myskin.SkinDetailHeaderScrollHelper;
import com.veganbeauty.app.utils.AddressBookHelper;

import java.util.List;

public class AccountProfileAddressFragment extends RootieFragment {

    private AccountProfileAddressBinding binding;
    private SkinDetailHeaderScrollHelper headerScrollHelper;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = AccountProfileAddressBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void setupUI(@NonNull View view) {
        Context context = requireContext();
        AddressBookHelper.ensureLoadedForCurrentUser(context);

        binding.btnBack.setOnClickListener(v -> getParentFragmentManager().popBackStack());
        binding.btnAddAddress.setOnClickListener(v -> showAddressDialog(null));

        loadAddressData();
        BottomNavHelper.setup(this, view, R.id.nav_account, tabId -> BottomNavHelper.navigate(this, tabId));
        setupScrollHideHeader();
    }

    private void setupScrollHideHeader() {
        float density = getResources().getDisplayMetrics().density;
        int bottomPadding = (int) (getResources().getDimension(R.dimen.home_nav_bar_height) + (76f * density));
        headerScrollHelper = new SkinDetailHeaderScrollHelper(
                binding.rlHeader,
                binding.scrollContent,
                bottomPadding
        );
        headerScrollHelper.attachToNestedScrollView(binding.scrollContent);
    }

    @Override
    public void onResume() {
        super.onResume();
        if (binding != null) {
            AddressBookHelper.ensureLoadedForCurrentUser(requireContext());
            loadAddressData();
        }
    }

    private void loadAddressData() {
        Context context = requireContext();
        List<AddressBookHelper.AddressEntry> list = AddressBookHelper.getSessionAddresses(context);
        LinearLayout container = binding.llAddressList;
        container.removeAllViews();

        if (list.isEmpty()) {
            binding.layoutEmpty.setVisibility(View.VISIBLE);
            container.setVisibility(View.GONE);
            return;
        }

        binding.layoutEmpty.setVisibility(View.GONE);
        container.setVisibility(View.VISIBLE);

        LayoutInflater inflater = LayoutInflater.from(context);
        for (AddressBookHelper.AddressEntry entry : list) {
            View item = inflater.inflate(R.layout.account_item_address, container, false);
            TextView tvLabel = item.findViewById(R.id.tv_label);
            TextView badgeDefault = item.findViewById(R.id.badge_default);
            TextView tvName = item.findViewById(R.id.tv_name);
            TextView tvPhone = item.findViewById(R.id.tv_phone);
            TextView tvAddress = item.findViewById(R.id.tv_address);
            ImageView ivIcon = item.findViewById(R.id.iv_label_icon);
            FrameLayout btnEdit = item.findViewById(R.id.btn_edit);
            FrameLayout btnDelete = item.findViewById(R.id.btn_delete);

            tvLabel.setText(entry.label);
            tvName.setText(entry.name);
            tvPhone.setText(entry.phone);
            tvAddress.setText(entry.address);
            applyLabelIcon(ivIcon, entry.label);

            if (entry.isDefault) {
                badgeDefault.setVisibility(View.VISIBLE);
                badgeDefault.setBackgroundResource(R.drawable.bg_dialog_btn_confirm);
                badgeDefault.setTextColor(Color.parseColor("#D9D9D9"));
            } else {
                badgeDefault.setVisibility(View.VISIBLE);
                badgeDefault.setBackgroundResource(R.drawable.tab_inactive_bg);
                badgeDefault.setTextColor(Color.parseColor("#000000"));
                badgeDefault.setOnClickListener(v -> {
                    AddressBookHelper.setDefaultById(context, entry.id);
                    loadAddressData();
                    Toast.makeText(context, "Đã đặt \"" + entry.label + "\" làm địa chỉ mặc định", Toast.LENGTH_SHORT).show();
                });
            }

            btnEdit.setOnClickListener(v -> showAddressDialog(entry));
            btnDelete.setOnClickListener(v -> new AlertDialog.Builder(context)
                    .setTitle("Xóa địa chỉ")
                    .setMessage("Bạn có chắc muốn xóa địa chỉ \"" + entry.label + "\"?")
                    .setPositiveButton("Xóa", (d, w) -> {
                        AddressBookHelper.deleteById(context, entry.id);
                        loadAddressData();
                        Toast.makeText(context, "Đã xóa địa chỉ", Toast.LENGTH_SHORT).show();
                    })
                    .setNegativeButton("Hủy", null)
                    .show());

            container.addView(item);
        }
    }

    private void applyLabelIcon(ImageView iv, String label) {
        String lower = label != null ? label.toLowerCase() : "";
        if (lower.contains("nhà") || lower.contains("home")) {
            iv.setImageResource(R.drawable.ic_home);
        } else if (lower.contains("văn phòng") || lower.contains("office") || lower.contains("công ty")) {
            iv.setImageResource(R.drawable.ic_address_office);
        } else {
            iv.setImageResource(R.drawable.ic_address_location_pin);
        }
    }

    private void showAddressDialog(@Nullable AddressBookHelper.AddressEntry existing) {
        Context context = requireContext();
        View dialogView = LayoutInflater.from(context).inflate(R.layout.account_dialog_edit_address, null);
        TextView tvTitle = dialogView.findViewById(R.id.tv_dialog_title);
        ChipGroup chipGroup = dialogView.findViewById(R.id.chip_group_label);
        EditText etLabel = dialogView.findViewById(R.id.et_dialog_label);
        EditText etName = dialogView.findViewById(R.id.et_dialog_name);
        EditText etPhone = dialogView.findViewById(R.id.et_dialog_phone);
        EditText etAddress = dialogView.findViewById(R.id.et_dialog_address);
        View btnCancel = dialogView.findViewById(R.id.btn_dialog_cancel);
        View btnSave = dialogView.findViewById(R.id.btn_dialog_save);

        boolean isEdit = existing != null;
        tvTitle.setText(isEdit ? "Chỉnh sửa địa chỉ" : "Thêm địa chỉ mới");

        String currentLabel = isEdit ? existing.label : "Nhà riêng";
        setupLabelChips(chipGroup, etLabel, currentLabel);

        if (isEdit) {
            etName.setText(existing.name);
            etPhone.setText(existing.phone);
            etAddress.setText(existing.address);
        } else {
            String sessionName = com.veganbeauty.app.data.local.ProfileSession.getFullName(context);
            String sessionPhone = com.veganbeauty.app.data.local.ProfileSession.getPhone(context);
            if (sessionName != null && !sessionName.trim().isEmpty()) etName.setText(sessionName.trim());
            if (sessionPhone != null && !sessionPhone.trim().isEmpty()) etPhone.setText(sessionPhone.trim());
        }

        AlertDialog dialog = new AlertDialog.Builder(context).setView(dialogView).create();
        if (dialog.getWindow() != null) {
            dialog.getWindow().setBackgroundDrawable(new android.graphics.drawable.ColorDrawable(Color.TRANSPARENT));
        }

        btnCancel.setOnClickListener(v -> dialog.dismiss());
        btnSave.setOnClickListener(v -> {
            String label = resolveSelectedLabel(chipGroup, etLabel);
            String name = etName.getText().toString().trim();
            String phone = etPhone.getText().toString().trim();
            String addr = etAddress.getText().toString().trim();
            if (label.isEmpty() || name.isEmpty() || phone.isEmpty() || addr.isEmpty()) {
                Toast.makeText(context, "Vui lòng nhập đầy đủ thông tin", Toast.LENGTH_SHORT).show();
                return;
            }
            if (isEdit) {
                existing.label = label;
                existing.name = name;
                existing.phone = phone;
                existing.address = addr;
                AddressBookHelper.updateAddress(context, existing);
                Toast.makeText(context, "Đã cập nhật địa chỉ", Toast.LENGTH_SHORT).show();
            } else {
                AddressBookHelper.addAddress(context, label, name, phone, addr, false);
                Toast.makeText(context, "Đã thêm địa chỉ mới", Toast.LENGTH_SHORT).show();
            }
            loadAddressData();
            dialog.dismiss();
        });

        dialog.show();
    }

    private void setupLabelChips(ChipGroup chipGroup, EditText etCustom, String currentLabel) {
        chipGroup.removeAllViews();
        int matchedChipId = View.NO_ID;
        for (String suggested : AddressBookHelper.SUGGESTED_LABELS) {
            Chip chip = new Chip(requireContext());
            chip.setText(suggested);
            chip.setCheckable(true);
            chip.setClickable(true);
            chipGroup.addView(chip);
            if (suggested.equalsIgnoreCase(currentLabel)) {
                matchedChipId = chip.getId();
            }
        }
        if (matchedChipId != View.NO_ID) {
            chipGroup.check(matchedChipId);
            etCustom.setVisibility(View.GONE);
        } else {
            for (int i = 0; i < chipGroup.getChildCount(); i++) {
                Chip chip = (Chip) chipGroup.getChildAt(i);
                if ("Khác".equals(chip.getText().toString())) {
                    chipGroup.check(chip.getId());
                    break;
                }
            }
            etCustom.setVisibility(View.VISIBLE);
            etCustom.setText(currentLabel);
        }

        chipGroup.setOnCheckedStateChangeListener((group, checkedIds) -> {
            if (checkedIds.isEmpty()) return;
            Chip selected = group.findViewById(checkedIds.get(0));
            if (selected != null && "Khác".equals(selected.getText().toString())) {
                etCustom.setVisibility(View.VISIBLE);
                etCustom.requestFocus();
            } else {
                etCustom.setVisibility(View.GONE);
            }
        });
    }

    @NonNull
    private String resolveSelectedLabel(ChipGroup chipGroup, EditText etCustom) {
        int checkedId = chipGroup.getCheckedChipId();
        if (checkedId != View.NO_ID) {
            Chip chip = chipGroup.findViewById(checkedId);
            if (chip != null) {
                String text = chip.getText().toString();
                if ("Khác".equals(text)) {
                    String custom = etCustom.getText().toString().trim();
                    return custom.isEmpty() ? "Khác" : custom;
                }
                return text;
            }
        }
        String custom = etCustom.getText().toString().trim();
        return custom.isEmpty() ? "Địa chỉ" : custom;
    }

    @Override
    public void observeViewModel() {
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/main/java/com/veganbeauty/app/features/profile/AccountProfileAddressFragment.kt
package com.veganbeauty.app.features.profile

import android.content.Context
import android.graphics.Color
import android.graphics.Typeface
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import com.veganbeauty.app.R
import com.veganbeauty.app.core.base.RootieFragment
import com.veganbeauty.app.data.local.ProfileSession
import com.veganbeauty.app.databinding.AccountProfileAddressBinding

class AccountProfileAddressFragment : RootieFragment() {

    private var _binding: AccountProfileAddressBinding? = null
    private val binding get() = _binding!!

    private val PREFS_NAME = "rootie_profile_prefs"
    
    // Preference Keys
    private val KEY_HOME_NAME = "addr_home_name"
    private val KEY_HOME_PHONE = "addr_home_phone"
    private val KEY_HOME_ADDR = "addr_home_addr"
    
    private val KEY_OFFICE_NAME = "addr_office_name"
    private val KEY_OFFICE_PHONE = "addr_office_phone"
    private val KEY_OFFICE_ADDR = "addr_office_addr"
    
    private val KEY_DEFAULT_TYPE = "addr_default_type" // "HOME" or "OFFICE"

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = AccountProfileAddressBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun setupUI(view: View) {
        val context = requireContext()

        // Back button action
        binding.btnBack.setOnClickListener {
            parentFragmentManager.popBackStack()
        }

        // Notification button
        binding.btnNotification.setOnClickListener {
            Toast.makeText(context, "Không có thông báo mới", Toast.LENGTH_SHORT).show()
        }

        // Load & bind data
        loadAddressData()

        // Badge / Card Default Click Listeners to toggle Default state
        binding.badgeHomeDefault.setOnClickListener {
            setDefaultAddress("HOME")
        }
        binding.badgeOfficeDefault.setOnClickListener {
            setDefaultAddress("OFFICE")
        }

        // Edit button clicks
        binding.btnHomeEdit.setOnClickListener {
            val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
            val currentName = prefs.getString(KEY_HOME_NAME, "Ánh Linh") ?: "Ánh Linh"
            val currentPhone = prefs.getString(KEY_HOME_PHONE, "0999 999 999") ?: "0999 999 999"
            val currentAddr = prefs.getString(KEY_HOME_ADDR, "123 Đường Bến Nghé, Phường Bến Nghé, TP.Hồ Chí Minh") ?: "123 Đường Bến Nghé, Phường Bến Nghé, TP.Hồ Chí Minh"
            
            showEditDialog("Nhà riêng", currentName, currentPhone, currentAddr) { name, phone, address ->
                prefs.edit().apply {
                    putString(KEY_HOME_NAME, name)
                    putString(KEY_HOME_PHONE, phone)
                    putString(KEY_HOME_ADDR, address)
                    apply()
                }
                // If this is default, sync with the central address
                if (prefs.getString(KEY_DEFAULT_TYPE, "HOME") == "HOME") {
                    ProfileSession.setAddress(context, address)
                }
                loadAddressData()
                Toast.makeText(context, "Đã cập nhật địa chỉ Nhà riêng", Toast.LENGTH_SHORT).show()
            }
        }

        binding.btnOfficeEdit.setOnClickListener {
            val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
            val currentName = prefs.getString(KEY_OFFICE_NAME, "Khánh Xuân") ?: "Khánh Xuân"
            val currentPhone = prefs.getString(KEY_OFFICE_PHONE, "0868 888 888") ?: "0868 888 888"
            val currentAddr = prefs.getString(KEY_OFFICE_ADDR, "Bitexco Financial Tower, 2 Hải Triều, Phường Bến Nghé, TP.Hồ Chí Minh") ?: "Bitexco Financial Tower, 2 Hải Triều, Phường Bến Nghé, TP.Hồ Chí Minh"
            
            showEditDialog("Văn phòng", currentName, currentPhone, currentAddr) { name, phone, address ->
                prefs.edit().apply {
                    putString(KEY_OFFICE_NAME, name)
                    putString(KEY_OFFICE_PHONE, phone)
                    putString(KEY_OFFICE_ADDR, address)
                    apply()
                }
                // If this is default, sync with the central address
                if (prefs.getString(KEY_DEFAULT_TYPE, "HOME") == "OFFICE") {
                    ProfileSession.setAddress(context, address)
                }
                loadAddressData()
                Toast.makeText(context, "Đã cập nhật địa chỉ Văn phòng", Toast.LENGTH_SHORT).show()
            }
        }

        // Delete button clicks (simulates deletion and resets values)
        binding.btnHomeDelete.setOnClickListener {
            AlertDialog.Builder(context)
                .setTitle("Xóa địa chỉ")
                .setMessage("Bạn có chắc chắn muốn xóa địa chỉ Nhà riêng?")
                .setPositiveButton("Xóa") { _, _ ->
                    val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
                    prefs.edit().apply {
                        remove(KEY_HOME_NAME)
                        remove(KEY_HOME_PHONE)
                        remove(KEY_HOME_ADDR)
                        // If it was default, switch default to Office
                        if (prefs.getString(KEY_DEFAULT_TYPE, "HOME") == "HOME") {
                            putString(KEY_DEFAULT_TYPE, "OFFICE")
                            val officeAddr = prefs.getString(KEY_OFFICE_ADDR, "Bitexco Financial Tower, 2 Hải Triều, Phường Bến Nghé, TP.Hồ Chí Minh")
                            if (officeAddr != null) {
                                ProfileSession.setAddress(context, officeAddr)
                            }
                        }
                        apply()
                    }
                    loadAddressData()
                    Toast.makeText(context, "Đã xóa địa chỉ Nhà riêng", Toast.LENGTH_SHORT).show()
                }
                .setNegativeButton("Hủy", null)
                .show()
        }

        binding.btnOfficeDelete.setOnClickListener {
            AlertDialog.Builder(context)
                .setTitle("Xóa địa chỉ")
                .setMessage("Bạn có chắc chắn muốn xóa địa chỉ Văn phòng?")
                .setPositiveButton("Xóa") { _, _ ->
                    val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
                    prefs.edit().apply {
                        remove(KEY_OFFICE_NAME)
                        remove(KEY_OFFICE_PHONE)
                        remove(KEY_OFFICE_ADDR)
                        // If it was default, switch default to Home
                        if (prefs.getString(KEY_DEFAULT_TYPE, "HOME") == "OFFICE") {
                            putString(KEY_DEFAULT_TYPE, "HOME")
                            val homeAddr = prefs.getString(KEY_HOME_ADDR, "123 Đường Bến Nghé, Phường Bến Nghé, TP.Hồ Chí Minh")
                            if (homeAddr != null) {
                                ProfileSession.setAddress(context, homeAddr)
                            }
                        }
                        apply()
                    }
                    loadAddressData()
                    Toast.makeText(context, "Đã xóa địa chỉ Văn phòng", Toast.LENGTH_SHORT).show()
                }
                .setNegativeButton("Hủy", null)
                .show()
        }

        // Add new address button click
        binding.btnAddAddress.setOnClickListener {
            showEditDialog("Mới", "", "", "") { name, phone, address ->
                // Simulate adding a third address or just setting/replacing one of the cards
                Toast.makeText(context, "Đã thêm địa chỉ mới thành công!", Toast.LENGTH_LONG).show()
            }
        }

        // Highlight "Tài khoản" active tab in bottom navigation shell
        view.findViewById<android.view.ViewGroup>(R.id.nav_account)?.let { navAccount ->
            val icon = navAccount.getChildAt(0) as? ImageView
            val label = navAccount.getChildAt(1) as? TextView
            icon?.setColorFilter(Color.parseColor("#677559"))
            label?.setTextColor(Color.parseColor("#677559"))
            label?.setTypeface(null, Typeface.BOLD)
        }
    }

    private fun loadAddressData() {
        val context = requireContext()
        val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)

        // Read or initialize Nhà riêng data
        val homeName = prefs.getString(KEY_HOME_NAME, "Ánh Linh") ?: "Ánh Linh"
        val homePhone = prefs.getString(KEY_HOME_PHONE, "0999 999 999") ?: "0999 999 999"
        val homeAddr = prefs.getString(KEY_HOME_ADDR, "123 Đường Bến Nghé, Phường Bến Nghé, TP.Hồ Chí Minh") ?: "123 Đường Bến Nghé, Phường Bến Nghé, TP.Hồ Chí Minh"

        // Read or initialize Văn phòng data
        val officeName = prefs.getString(KEY_OFFICE_NAME, "Khánh Xuân") ?: "Khánh Xuân"
        val officePhone = prefs.getString(KEY_OFFICE_PHONE, "0868 888 888") ?: "0868 888 888"
        val officeAddr = prefs.getString(KEY_OFFICE_ADDR, "Bitexco Financial Tower, 2 Hải Triều, Phường Bến Nghé, TP.Hồ Chí Minh") ?: "Bitexco Financial Tower, 2 Hải Triều, Phường Bến Nghé, TP.Hồ Chí Minh"

        val defaultType = prefs.getString(KEY_DEFAULT_TYPE, "HOME") ?: "HOME"

        // Bind Card 1 UI
        binding.tvHomeName.text = homeName
        binding.tvHomePhone.text = homePhone
        binding.tvHomeAddress.text = homeAddr

        // Bind Card 2 UI
        binding.tvOfficeName.text = officeName
        binding.tvOfficePhone.text = officePhone
        binding.tvOfficeAddress.text = officeAddr

        // Bind Badge styling dynamically according to Default status
        if (defaultType == "HOME") {
            // Card Home is DEFAULT (Active)
            binding.badgeHomeDefault.setBackgroundResource(R.drawable.bg_badge_default_active)
            binding.badgeHomeDefault.setTextColor(Color.parseColor("#D9D9D9"))
            
            // Card Office is NOT DEFAULT (Inactive)
            binding.badgeOfficeDefault.setBackgroundResource(R.drawable.bg_badge_default_inactive)
            binding.badgeOfficeDefault.setTextColor(Color.parseColor("#000000"))
        } else {
            // Card Office is DEFAULT (Active)
            binding.badgeOfficeDefault.setBackgroundResource(R.drawable.bg_badge_default_active)
            binding.badgeOfficeDefault.setTextColor(Color.parseColor("#D9D9D9"))
            
            // Card Home is NOT DEFAULT (Inactive)
            binding.badgeHomeDefault.setBackgroundResource(R.drawable.bg_badge_default_inactive)
            binding.badgeHomeDefault.setTextColor(Color.parseColor("#000000"))
        }
    }

    private fun setDefaultAddress(type: String) {
        val context = requireContext()
        val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
        
        prefs.edit().putString(KEY_DEFAULT_TYPE, type).apply()
        
        // Sync the main address stored in ProfileSession to the default one chosen
        val defaultAddr = if (type == "HOME") {
            prefs.getString(KEY_HOME_ADDR, "123 Đường Bến Nghé, Phường Bến Nghé, TP.Hồ Chí Minh")
        } else {
            prefs.getString(KEY_OFFICE_ADDR, "Bitexco Financial Tower, 2 Hải Triều, Phường Bến Nghé, TP.Hồ Chí Minh")
        }
        
        if (defaultAddr != null) {
            ProfileSession.setAddress(context, defaultAddr)
        }

        loadAddressData()
        val targetName = if (type == "HOME") "Nhà riêng" else "Văn phòng"
        Toast.makeText(context, "Đã đặt $targetName làm địa chỉ mặc định", Toast.LENGTH_SHORT).show()
    }

    private fun showEditDialog(
        title: String,
        currentName: String,
        currentPhone: String,
        currentAddress: String,
        onSave: (name: String, phone: String, address: String) -> Unit
    ) {
        val dialogView = LayoutInflater.from(context).inflate(R.layout.account_dialog_edit_address, null)
        val tvTitle = dialogView.findViewById<TextView>(R.id.tv_dialog_title)
        val etName = dialogView.findViewById<EditText>(R.id.et_dialog_name)
        val etPhone = dialogView.findViewById<EditText>(R.id.et_dialog_phone)
        val etAddress = dialogView.findViewById<EditText>(R.id.et_dialog_address)
        val btnCancel = dialogView.findViewById<View>(R.id.btn_dialog_cancel)
        val btnSave = dialogView.findViewById<View>(R.id.btn_dialog_save)

        tvTitle.text = "Chỉnh sửa địa chỉ: $title"
        etName.setText(currentName)
        etPhone.setText(currentPhone)
        etAddress.setText(currentAddress)

        val dialog = AlertDialog.Builder(requireContext())
            .setView(dialogView)
            .create()

        dialog.window?.setBackgroundDrawable(android.graphics.drawable.ColorDrawable(Color.TRANSPARENT))

        btnCancel.setOnClickListener {
            dialog.dismiss()
        }

        btnSave.setOnClickListener {
            val name = etName.text.toString().trim()
            val phone = etPhone.text.toString().trim()
            val addr = etAddress.text.toString().trim()
            if (name.isNotEmpty() && phone.isNotEmpty() && addr.isNotEmpty()) {
                onSave(name, phone, addr)
                dialog.dismiss()
            } else {
                Toast.makeText(context, "Vui lòng nhập đầy đủ thông tin", Toast.LENGTH_SHORT).show()
            }
        }

        dialog.show()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}
